core/src/main/java/com/datastrato/gravitino/tenant/AccessControlManager.java

```diff
@@ -0,0 +1,134 @@
+import com.datastrato.gravitino.exceptions.NoSuchUserException;
+import com.datastrato.gravitino.exceptions.UserAlreadyExistsException;
+import com.datastrato.gravitino.meta.AuditInfo;
+import com.datastrato.gravitino.meta.MetalakeUser;
+import com.datastrato.gravitino.storage.IdGenerator;
+import com.datastrato.gravitino.utils.PrincipalUtils;
+import java.io.IOException;
+import java.time.Instant;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/* AccessControlManager is used for manage users, roles, grant information, this class is
+ * an important class for tenant management. */
+public class AccessControlManager implements SupportsUserManagement {
+
+  private static final String USER_DOES_NOT_EXIST_MSG = "User %s does not exist in th metalake %s";
+
+  private static final Logger LOG = LoggerFactory.getLogger(AccessControlManager.class);
+
+  private final EntityStore store;
+  private final IdGenerator idGenerator;
+
+  /**
+   * Constructs a AccessControlManager instance.
+   *
+   * @param store The EntityStore to use for managing access control.
+   * @param idGenerator The IdGenerator to use for generating identifiers.
+   */
+  public AccessControlManager(EntityStore store, IdGenerator idGenerator) {
+    this.store = store;
+    this.idGenerator = idGenerator;
+  }
+
+  /**
+   * Creates a new User.
+   *
+   * @param metalake The Metalake of the User.
+   * @param userName THe name of the User.
+   * @param properties Additional properties for the Metalake.
+   * @return The created User instance.
+   * @throws UserAlreadyExistsException If a User with the same identifier already exists.
+   * @throws RuntimeException If creating the User encounters storage issues.
+   */
+  @Override
+  public User createUser(String metalake, String userName, Map<String, String> properties)
+      throws UserAlreadyExistsException {
+    MetalakeUser metalakeUser =
+        MetalakeUser.builder()
+            .withId(idGenerator.nextId())
+            .withName(userName)
+            .withMetalake(metalake)
+            .withProperties(properties)
+            .withAuditInfo(
+                AuditInfo.builder()
+                    .withCreator(PrincipalUtils.getCurrentPrincipal().getName())
+                    .withCreateTime(Instant.now())
+                    .build())
+            .build();
+    try {
+      store.put(metalakeUser, false /* overwritten */);
+      return metalakeUser;
+    } catch (EntityAlreadyExistsException e) {
+      LOG.warn("User {} in the metalake {} already exists", userName, metalake, e);
+      throw new UserAlreadyExistsException(
+          "User %s in the metalake %s already exists", userName, metalake);
+    } catch (IOException ioe) {
+      LOG.error(
+          "Creating user {} failed in the metalake {} due to storage issues",
+          userName,
+          metalake,
+          ioe);
+      throw new RuntimeException(ioe);
+    }
+  }
+
```

Comment:
We don't implement the `AlterUser`. `AlterUser` should change some properties. I will create a good first issue for new comers.